uniprotkb-rest/src/test/java/org/uniprot/api/uniprotkb/queue/AsyncDownloadIntegTest.java

```diff
@@ -0,0 +1,343 @@
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.boot.test.autoconfigure.web.client.AutoConfigureWebClient;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.SpyBean;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.uniprot.api.common.repository.solrstream.FacetTupleStreamTemplate;
+import org.uniprot.api.common.repository.stream.common.TupleStreamTemplate;
+import org.uniprot.api.rest.download.AsyncDownloadTestConfig;
+import org.uniprot.api.rest.download.configuration.RedisConfiguration;
+import org.uniprot.api.rest.download.model.DownloadJob;
+import org.uniprot.api.rest.download.model.DownloadRequestToArrayConverter;
+import org.uniprot.api.rest.download.model.HashGenerator;
+import org.uniprot.api.rest.download.model.JobStatus;
+import org.uniprot.api.rest.download.queue.ProducerMessageService;
+import org.uniprot.api.rest.download.repository.DownloadJobRepository;
+import org.uniprot.api.rest.output.context.FileType;
+import org.uniprot.api.rest.request.DownloadRequest;
+import org.uniprot.api.uniprotkb.UniProtKBREST;
+import org.uniprot.api.uniprotkb.controller.AbstractUniProtKBDownloadIT;
+import org.uniprot.api.uniprotkb.controller.UniProtKBDownloadController;
+import org.uniprot.api.uniprotkb.controller.request.UniProtKBDownloadRequest;
+import org.uniprot.api.uniprotkb.repository.DataStoreTestConfig;
+import org.uniprot.api.uniprotkb.repository.store.UniProtStoreConfig;
+import org.uniprot.store.search.SolrCollection;
+
+import com.jayway.jsonpath.JsonPath;
+
+@ActiveProfiles(profiles = {"offline", "asyncDownload"})
+@EnableConfigurationProperties
+@PropertySource("classpath:application.properties")
+@ContextConfiguration(
+        classes = {
+            DataStoreTestConfig.class,
+            UniProtKBREST.class,
+            UniProtStoreConfig.class,
+            AsyncDownloadTestConfig.class,
+            RedisConfiguration.class
+        })
+@ExtendWith(SpringExtension.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@WebMvcTest({UniProtKBDownloadController.class})
+@AutoConfigureWebClient
+public class AsyncDownloadIntegTest extends AbstractUniProtKBDownloadIT {
+    @Autowired private TupleStreamTemplate tupleStreamTemplate;
+    @Autowired private FacetTupleStreamTemplate facetTupleStreamTemplate;
+
+    @Autowired
+    @Qualifier("uniProtKBSolrClient")
+    private SolrClient
+            solrClient; // this is NOT inmemory solr cluster client, see CloudSolrClient in parent
+    // class
+
+    @SpyBean @Autowired
```

Comment:
Why you want to use @Autowired and @SpyBean together?

Reply:
thanks @supun-ebi we need only @SpyBean annotation.